### PR TITLE
add additional hive config for xgboost codegen

### LINKED ISSUE #728: Can't use XGBoost and TensorFlow at the same time.
It seems that executing the train/predict SQL in the Notebook would cause an error, and can not use XGBoost and Tensroflow at the same time.

If using the default `sqlflow_submitter` which is suitable with Tensorflow, the behavior of using XGBoost will cause sqlflow to throw exception like 

```
sqlflow_submitter.xgboost.common.XGBoostError: XGBoost training task failed: TProtocolException(type=4)

StatusCode.UNKNOWN
training failed exit status 1
```

This problem  can be solved by setting environment variable `SQLFLOW_submitter=xgboost`. However, this setting is in conflict with TensorFlow.

Is there any way to solve this problem?

### PR DESCRIPTION
**Pull request by sperlingxx**
fix #728

### CODE REVIEW

**Review by weiguoz — APPROVED** [review]
LGTM

File: sql/codegen.go [hunk 1]
```diff
@@ -246,6 +232,8 @@ func fillDatabaseInfo(r *filler, db *DB) (*filler, error) {
 		if err != nil {
 			return nil, err
 		}
+		r.Auth = cfg.Auth
+		r.Session = cfg.SessionCfg
```
Comment (weiguoz) [inline sql/codegen.go:236]: Nice practice!

**Review by weiguoz — DISMISSED** [review]

File: sql/python/sqlflow_submitter/xgboost/sqlflow_data_source.py [hunk 2]
```diff
@@ -115,7 +115,11 @@ def __init__(self, rank: int, num_worker: int,
             self._result_schema = {'append_columns': column_conf.append_columns or []}
             self._result_schema.update(column_conf.result_columns._asdict())
 
-        conn = connect(**source_conf.db_config)
+        # assert sqlflow_submitter.db.connect contains no varargs
+        import inspect
```
Comment (terrytangyuan) [inline sql/python/sqlflow_submitter/xgboost/sqlflow_data_source.py]: Move this to top of the file

Comment (sperlingxx) [reply sql/python/sqlflow_submitter/xgboost/sqlflow_data_source.py]: @terrytangyuan Done

**Review by weiguoz — APPROVED** [review]

**Pull request merged**